File: Data Structures & Algorithms/reverse-integer/submission-1.py
class Solution:
    def reverse(self, x: int) -> int:

        sgn = 1
        if x < 0:
            sgn = -1
            x *= -1

        res = 0
        while x:
            res = (res + (x % 10)) * 10
            x = int(x / 10)

        if res - ((1 << 32) - 1) > 0:
            return 0

        return sgn * int(res / 10)

File: Data Structures & Algorithms/sum-of-two-integers/submission-1.java
class Solution {
    public int getSum(int a, int b) {

        int res = 0;
        int c = 0;
        for(int i = 0; i < 32; i++){
            int[] calList = myCal(c, (a & 1), (b & 1));
            c = calList[0];
            int s = calList[1];
            calList = null;
            res = res | (s << i);
            a = a >> 1;
            b = b >> 1;
        }
        return res;

    }

    int[] myCal(int c, int x, int y){
        int s = c ^ (x ^ y);
        c = (x & y) | (c & (x ^ y));
        // int s = c ^ (x ^ y) ORDER MATTRES
        int[] res = {c, s};
        return res;
    }

}
